### PR TITLE
SW-160 Masterpass

### PR DESCRIPTION
**Pull request by fatchip-Stefan**

### CODE REVIEW

File: Frontend/MoptPaymentPayone/Components/Classes/PayonePaymentHelper.php [hunk 1]
```diff
@@ -783,6 +797,21 @@ public function isPayoneAmazonPay($paymentName)
         }
     }
 
+    /**
+     * check if given payment name is payone amazonpay
```
Comment (fjbender) [inline Frontend/MoptPaymentPayone/Components/Classes/PayonePaymentHelper.php]: *masterpass

File: Frontend/MoptPaymentPayone/Components/Payone/Api/Enum/GenericpaymentAction.php [hunk 2]
```diff
@@ -42,5 +42,9 @@ class Payone_Api_Enum_GenericpaymentAction
 
     const AMAZON_GETORDERREFERENCEDETAILS = "getorderreferencedetails";
 
-    const AMAZON_SETORDERREFERENCEDETAILS = "setorderreferencedetails";
```
Comment (fjbender) [inline Frontend/MoptPaymentPayone/Components/Payone/Api/Enum/GenericpaymentAction.php]: Is this really not needed anymore?

File: Frontend/MoptPaymentPayone/Controllers/Frontend/FatchipBSPayoneMasterpass.php [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+    public function indexAction()
+    {
+        $this->forward('gateway');
+    }
+
+    public function errorAction()
+    {
+        $params = $this->Request()->getParams();
+        die(var_dump($params));
```
Comment (fjbender) [inline Frontend/MoptPaymentPayone/Controllers/Frontend/FatchipBSPayoneMasterpass.php]: Is this an appropriate action here?

File: Frontend/MoptPaymentPayone/Controllers/Frontend/FatchipBSPayoneMasterpass.php [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+                $addressData = $response->getPayDataArray();
+
+                $this->forward('register', 'FatchipBSPayoneMasterpassRegister', null, ['BSPayoneAddressData' => $addressData]);
+                break;
+            default:
+                $this->forward('error');
+                break;
+        }
+
+        // Steftest
```
Comment (fjbender) [inline Frontend/MoptPaymentPayone/Controllers/Frontend/FatchipBSPayoneMasterpass.php]: This should probably be removed?

File: Frontend/MoptPaymentPayone/Controllers/Frontend/FatchipBSPayoneMasterpass.php [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+            $this->service = $this->payoneServiceBuilder->buildServicePaymentAuthorize();
+            $this->session->moptIsAuthorized = true;
+        }
+
+        $request->setWorkorderId($this->session->offsetGet('fatchipBSPayoneMasterPassWorkOrderId'));
+        $request->setClearingtype($clearingType);
+        $request->setWallettype($walletType);
+        $request->setCurrency("EUR");
+        $request->setAmount($orderVariables['sAmount']);
+        // TODO: use order number
```
Comment (fjbender) [inline Frontend/MoptPaymentPayone/Controllers/Frontend/FatchipBSPayoneMasterpass.php]: This should probably be the order number

File: Frontend/MoptPaymentPayone/Controllers/Frontend/FatchipBSPayoneMasterpassRegister.php [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+        $addressData['firstname'] = utf8_decode($addressData['firstname']);
+        $addressData['lastname'] = utf8_decode($addressData['lastname']);
+        $addressData['street'] = utf8_decode($addressData['street']);
+        $addressData['shipping_street'] = utf8_decode($addressData['shipping_street']);
+
+            // StefTEst Remove
+        $testBefore = $session->offsetGet('sPaymentID');
+
+        $session->offsetSet('sPaymentID', $this->moptPayonePaymentHelper->getPaymentIdFromName('mopt_payone__ewallet_masterpass'));
+        // StefTEst RovePayone Connector
```
Comment (fjbender) [inline Frontend/MoptPaymentPayone/Controllers/Frontend/FatchipBSPayoneMasterpassRegister.php]: Is this still needed?

File: Frontend/MoptPaymentPayone/Controllers/Frontend/FatchipBSPayoneMasterpassRegister.php [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+        $testAfter = $session->offsetGet('sPaymentID');
+        // set flag so we do not get edirected back to shippingpayment
+        $session->offsetSet('moptFormSubmitted', true);
+
+        $this->view->assign('fatchipBSPayone', $addressData);
+        $this->view->loadTemplate('frontend/fatchipBSPayoneMasterpassRegister/index.tpl');
+    }
+
+
+    // TODO this should not be neccessary but somehow the parent os not
```
Comment (fjbender) [inline Frontend/MoptPaymentPayone/Controllers/Frontend/FatchipBSPayoneMasterpassRegister.php]: Still a todo?

File: Frontend/MoptPaymentPayone/Views/frontend/_resources/javascript/fatchipBSPayoneMasterpass.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        zip2: false,
+        city2: false,
+        countryCodeShippingID: false
+    },
+
+    init: function () {
+        "use strict";
+        var me = this;
+        me.applyDataAttributes();
+        // TESTING
```
Comment (fjbender) [inline Frontend/MoptPaymentPayone/Views/frontend/_resources/javascript/fatchipBSPayoneMasterpass.js]: Should we remove this in production?

File: Frontend/MoptPaymentPayone/Views/frontend/checkout/ajax_cart_masterpass.tpl [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+        <div class="clear"></div>
+    </div>
+    <script>
+        function getMasterPassData(){
+
+            $.getScript('https://sandbox.masterpass.com/lightbox/Switch/integration/MasterPass.client.js');
+            $.post('{url controller="moptAjaxPayone" action="buildAndCallSetCheckout" forceSecure}', function (response)
+            {
+                var data = $.parseJSON(response);
+                console.log("token:");
```
Comment (fjbender) [inline Frontend/MoptPaymentPayone/Views/frontend/checkout/ajax_cart_masterpass.tpl]: console.log should probably be removed in production

File: Frontend/MoptPaymentPayone/Views/frontend/checkout/mopt_cart_masterpass.tpl [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        <div class="clear"></div>
+    </div>
+    <script>
+        function getMasterPassData(){
+
+            $.getScript('https://sandbox.masterpass.com/lightbox/Switch/integration/MasterPass.client.js');
+            $.post('{url controller="moptAjaxPayone" action="buildAndCallSetCheckout" forceSecure}', function (response)
+            {
+                var data = $.parseJSON(response);
+                console.log("token:");
```
Comment (fjbender) [inline Frontend/MoptPaymentPayone/Views/frontend/checkout/mopt_cart_masterpass.tpl]: See above

File: Frontend/MoptPaymentPayone/Components/Payone/Api/Enum/GenericpaymentAction.php (hunk 2 above)
Comment (fatchip-Stefan) [reply Frontend/MoptPaymentPayone/Components/Payone/Api/Enum/GenericpaymentAction.php]: thanks for pointing this out

File: Frontend/MoptPaymentPayone/Views/frontend/_resources/javascript/fatchipBSPayoneMasterpass.js (hunk 8 above)
Comment (fatchip-Stefan) [reply Frontend/MoptPaymentPayone/Views/frontend/_resources/javascript/fatchipBSPayoneMasterpass.js]: thx, i removed it

**Pull request merged**